src/main/kotlin/ch/bfh/habits/controllers/AuthController.kt

```
@@ -19,34 +19,66 @@
+        val user = userService.findByUserName(userName)
+        println("Fetched user details: $user")
+        return ResponseEntity.ok(user)
+    }
+
+    @DeleteMapping("api/user/{userName}")
+    fun deleteUser(@PathVariable userName: String): ResponseEntity<Any?> {
+        return try {
+            userService.deleteByUserName(userName)
+            println("User deleted: $userName")
+            ResponseEntity.status(200).body("User deleted successfully")
+        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)